src/api/controller/api/characteristic.js

```diff
@@ -18,14 +18,48 @@ const isValueField = field =>
     get(field, 'transformers[0].operation') === 'VALUE';
 
 const updateFieldValue = async (ctx, field, value) => {
+    if (get(field, 'transformers[0].args[0].value') === value) {
+        return field;
+    }
+
     // lodash.set mutate the object, so we need to copy the object first
     const copiedField = deepCopy(field);
-    return await ctx.field.updateOneById(
+    return ctx.field.updateOneById(
         field._id,
         set(copiedField, 'transformers[0].args[0].value', value),
     );
 };
 
+const updateField = (ctx, requestedNewCharacteristics) => async field => {
+    const body = {};
+
+    if (isValueField(field)) {
```

Comment:
how about adding a getFieldType function and then using switch ?

Reply:
I can't. What about the field is both VALUE and composedOf?